src/lib/sql/sql_translator.cpp

```diff
@@ -377,46 +393,95 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
+  return join_node;
+}
 
-  Assert(static_cast<bool>(left_in_left_node) ^ static_cast<bool>(left_in_right_node),
-         std::string("Left operand ") + left_qualified_column_name.as_string() +
-             " must be in exactly one of the input nodes");
-  Assert(static_cast<bool>(right_in_left_node) ^ static_cast<bool>(right_in_right_node),
-         std::string("Right operand ") + right_qualified_column_name.as_string() +
-             " must be in exactly one of the input nodes");
+LQPInputSide SQLTranslator::_get_side(const std::shared_ptr<AbstractLQPNode>& left_node,
+                                      const std::shared_ptr<AbstractLQPNode>& right_node, const hsql::Expr* expr) {
+  DebugAssert(expr && expr->type == hsql::kExprColumnRef, "only applicable to columnar expressions");
+  const auto qualified = HSQLExprTranslator::to_qualified_column_name(*expr);
+  const auto left = left_node->find_column(qualified);
+  const auto right = right_node->find_column(qualified);
+  Assert(left.has_value() ^ right.has_value(),
+         std::string("Column '") + qualified.as_string() + "' was not found or is not unique");
+  return left ? LQPInputSide::Left : LQPInputSide::Right;
+}
 
-  const auto column_references = left_in_left_node ? std::make_pair(*left_in_left_node, *right_in_right_node)
-                                                   : std::make_pair(*left_in_right_node, *right_in_left_node);
+void SQLTranslator::_get_sides_from_operator_expressions(std::vector<const hsql::Expr*>& operators,
+                                                         std::shared_ptr<AbstractLQPNode>& left_node,
+                                                         std::shared_ptr<AbstractLQPNode>& right_node,
+                                                         std::vector<const hsql::Expr*>& both,
+                                                         std::vector<const hsql::Expr*>& left,
+                                                         std::vector<const hsql::Expr*>& right) {
+  for (const auto* op : operators) {
+    const bool left_is_column = op->expr && op->expr->type == hsql::kExprColumnRef;
+    const bool right_is_column = op->expr2 && op->expr2->type == hsql::kExprColumnRef;
+
+    if (left_is_column && right_is_column &&
+        _get_side(left_node, right_node, op->expr) != _get_side(left_node, right_node, op->expr2)) {
+      both.push_back(op);
+      continue;
+    }
 
-  // Joins currently only support one simple condition (i.e., not multiple conditions).
-  auto predicate_condition = translate_operator_type_to_predicate_condition(condition.opType);
+    if (left_is_column) {
+      auto& side = (_get_side(left_node, right_node, op->expr) == LQPInputSide::Left) ? left : right;
+      side.push_back(op);
+    } else if (right_is_column) {
+      auto& side = (_get_side(left_node, right_node, op->expr2) == LQPInputSide::Left) ? left : right;
+      side.push_back(op);
+    } else {
+      left.push_back(op);
```

Comment:
This looks like you would push `1=2` to the left side? I would rather throw an exception for that.

Reply:
OK better be a little more defensive albeit lacking columns in join conditions does not seem to violate the SQL spec.

Reply:
No, probably you are not missing anything here. AFAIK a condition without any columns does not make much sense. As shown in your example only literals would remain, which either evaluate to true or false independently of any input.

Reply:
It doesn’t make sense, but it is valid SQL and could be generated by OOMs (brrr)

Reply:
Detecting likely misuse of the method should be preferred over support for faulty ORMs.